src/native/libs/System.Native/pal_networking.c

```diff
@@ -1588,6 +1588,50 @@ int32_t SystemNative_Connect(intptr_t socket, uint8_t* socketAddress, int32_t so
     return err == 0 ? Error_SUCCESS : SystemNative_ConvertErrorPlatformToPal(errno);
 }
 
+int32_t SystemNative_Connectx(intptr_t socket, uint8_t* socketAddress, int32_t socketAddressLen, uint8_t* data, int32_t dataLen, int32_t tfo, int* sent)
+{
+    if (socketAddress == NULL || socketAddressLen < 0 || sent == NULL)
+    {
+        return Error_EFAULT;
+    }
+
+    int fd = ToFileDescriptor(socket);
+#if HAVE_CONNECTX
+    struct sa_endpoints eps;
+    struct iovec iovec;
+    int flags = 0;
+    iovec.iov_base = data;
+    iovec.iov_len = dataLen;
+    memset(&eps, 0, sizeof(eps));
+    eps.sae_dstaddr = (struct sockaddr *)socketAddress;
+    eps.sae_dstaddrlen = socketAddressLen;
+
+    size_t length = 0;
+    int err;
+    while ((err = connectx(fd, &eps, SAE_ASSOCID_ANY, tfo != 0 ? CONNECT_DATA_IDEMPOTENT : 0, dataLen > 0 ? &iovec : NULL, dataLen > 0 ? 1 : 0, &length, NULL)) < 0 && errno == EINTR);
+    *sent = (int)length;
+
+    return err == 0 ? Error_SUCCESS : SystemNative_ConvertErrorPlatformToPal(errno);
+#else
+#ifdef TCP_FASTOPEN_CONNECT
+    int enabled = 1;
+    socklen_t len = sizeof(enabled);
+
+    if (tfo)
+    {
+        // This will either success and connect will finish without sending SYN until we write to so the socket.
+        // If this is not available we simply connect and write provided data afterwards.
+        setsockopt(fd, IPPROTO_TCP, TCP_FASTOPEN_CONNECT, &enabled, len);
+    }
+#endif
+    (void*)data;
+    (void)dataLen;
+    (void)tfo;
+    sent = 0;
```

Comment:
???

Reply:
unlike `connectex` and `connects` we do not sent any data here. e.g. even if provided we process 0 bytes.

Reply:
Does

```
    (void*)data;
    (void)dataLen;
    (void)tfo;
```
do anything meaningful here?

Reply:
avoids compiler warning about unused parameters.

Reply:
It needs a comment then.

Reply:
updated. We use the parent in other parts of the code but extra comment never hurts.